### PR TITLE
[PE1-1248]  Add repository to manage Access Origin Control (AOC)

### PR DESCRIPTION
**Pull request by LCaparelli**
## Description

TSIA

## Motivation and Context

Necessary so that we can eventually support direct access to private S3 buckets that are not set up with static website hosting as origin.

## How has this been tested?

Automated tests.

## Checklist

- [x] I have implemented automated tests for the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by go-faustino — APPROVED** [review]
LGTM!!! 🎸 

Left a really minor typo nit, but that's it.

**Review by caiofralmeida — APPROVED** [review]
I left an optional suggestion, but LGTM :rocket:

File: internal/aws/error.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+package aws
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+// IsErrorCode returns whether the give error matches an AWS error with the give errCode.
```
[the remark this answers is not in the document]
Comment (LCaparelli) [reply internal/aws/error.go:29]: Nice catch, thanks!

File: internal/cloudfront/aoc_repository.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudfront "github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/aws/aws-sdk-go/service/cloudfront/cloudfrontiface"
+
+	cdnaws "github.com/Gympass/cdn-origin-controller/internal/aws"
+)
+
+var errNoSuchAOC = errors.New("aoc does not exist")
+
+type AOC struct {
```
[the remark this answers is not in the document]
Comment (LCaparelli) [reply internal/cloudfront/aoc_repository.go:35]: +1, already did it on a branch for an upcoming PR #81, thanks!

**Pull request merged**